src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -272,9 +277,12 @@ private String getConstantString(Class<?> type, Object value) {
                 buf.append("d");
             } else {
                 buf.append("Double.valueOf(\"");
-                buf.append(value.toString());
+                buf.append(value);
                 buf.append("\")");
             }
+        } else if (type == boolean.class) {
```

Comment:
This is a bug that was revealed by the cleanup: for boolean constants macros, we were using a "boolean" layout - but then we were dropping that information away, and just using byte as a carrier. That's because the Type -> Layout translation was subtly different from the Type -> Class one. Now that we only use a single translation, all these discrepancies are exposed.